### PR TITLE
fixed multiple sharePopup bug.

### PR DESCRIPTION
**Pull request by AslamThachapalli**
**Issue:**
- On multiple clicks on share actions icon, multiple popup was getting shown.

**SS**

https://github.com/Sangwan5688/BlackHole/assets/113328135/554bf0a3-6390-4fba-a3e7-62232f55a8c0

*Fixed* this issue. 
- The corrections make sure only one popup will be shown at a time.

### CODE REVIEW

File: lib/Screens/Common/song_list.dart [hunk 1]
```diff
@@ -229,9 +233,21 @@ class _SongsListPageState extends State<SongsListPage> {
...
+                            if (!isSharePopupShown) {
+                              setState(() {
+                                isSharePopupShown = true;
+                              });
+
+                              Share.share(
+                                widget.listItem['perma_url'].toString(),
+                              ).whenComplete(() {
+                                Timer(const Duration(milliseconds: 500), () {
+                                  setState(() {
```
Comment (Sangwan5688) [inline lib/Screens/Common/song_list.dart:245]: is setState even required for this?

File: lib/Screens/Player/audioplayer.dart [hunk 2]
```diff
@@ -269,10 +271,22 @@ class _PlayScreenState extends State<PlayScreen> {
                       IconButton(
                         icon: const Icon(Icons.share_rounded),
                         tooltip: AppLocalizations.of(context)!.share,
-                        onPressed: () {
-                          Share.share(
-                            mediaItem.extras!['perma_url'].toString(),
-                          );
+                        onPressed: () async {
+                          if (!isSharePopupShown) {
+                            setState(() {
```
Comment (Sangwan5688) [inline lib/Screens/Player/audioplayer.dart:276]: same question here as well

File: lib/Screens/Common/song_list.dart (hunk 1 above)
Comment (AslamThachapalli) [reply lib/Screens/Common/song_list.dart]: Yeah true. setState is not required. my bad :)

File: lib/Screens/Player/audioplayer.dart (hunk 2 above)
Comment (AslamThachapalli) [reply lib/Screens/Player/audioplayer.dart]: Yeah, here as well.

**Pull request merged**